app/models/agents/phantom_js_cloud_agent.rb

```diff
@@ -0,0 +1,167 @@
+require 'json'
+require 'uri'
+
+include ERB::Util
+
+
+module Agents
+  class PhantomJsCloudAgent < Agent
+    include FormConfigurable
+    include WebRequestConcern
+
+    can_dry_run!
+
+    default_schedule "every_12h"
+
+    description <<-MD
+      [PhantomJs Cloud](https://phantomjscloud.com/) renders webpages in much the same way as a browser would, and allows the Website Agent to properly scrape dynamic content from javascript-heavy pages.
+
+      The Phantom Js Cloud Agent is used to formulate a url in accordance with the [PhantomJs Cloud API](https://phantomjscloud.com/docs/index.html).
+      This url can then be supplied to Website Agent to fetch and parse content.
+
+      [Sign up](https://dashboard.phantomjscloud.com/dash.html#/signup) to get an api key, and add it in Huginn credentials.
+
+
+      Options:
+
+      * `Api key` - PhantomJs Cloud API Key credential stored in Huginn
+      * `Url` - The url to render
+      * `Mode` - Create a new `clean` event or `merge` old payload with new values (default: `clean`)
+      * `Render type` - Render as html or plain text without html tags (default: `html`)
+      * `Output as json` - Return the page conents and metadata as a JSON object (default: `false`)
+      * `Ignore images` - Skip loading of inlined images (default: `false`)
+      * `Url agent` - A custom User-Agent name (default: `#{default_user_agent}`)
+      * `Wait interval` - Milliseconds to delay rendering after the last resource is finished loading.
+      This is useful in case there are any AJAX requests or animations that need to finish up.
+      This can safely be set to 0 if you know there are no AJAX or animations you need to wait for (default: `1000`ms)
+
+    MD
+
+    event_description <<-MD
+      Events look like this:
+          {
+            "url": "..."
+          }
+    MD
+
+    def default_options
+      {
+          'mode' => 'clean',
+          'url' => "http://xkcd.com",
+          'render_type' => "html",
+          'output_as_json' => false,
+          'ignore_images' => false,
+          'user_agent' => self.class.default_user_agent,
+          'wait_interval' => "1000"
+      }
+    end
+
+    form_configurable :mode, type: :array, values: ['clean', 'merge']
+    form_configurable :api_key, roles: :completable
+    form_configurable :url
+    form_configurable :render_type, type: :array, values: ['html', 'plainText']
+    form_configurable :output_as_json, type: :boolean
+    form_configurable :ignore_images, type: :boolean
+    form_configurable :user_agent, type: :text
+    form_configurable :wait_interval
+
+    def mode
+      interpolated['mode'].presence || default_options['mode']
+    end
+
+    def renderType
```

Comment:
Ruby style is generally underscores for variable and method names, so `render_type`, not `renderType`.